components/tidb_query/src/batch/runner.rs

```diff
@@ -393,29 +393,29 @@ impl<SS: 'static> BatchExecutorsRunner<SS> {
                     // Although `schema()` can be deeply nested, it is ok since we process data in
                     // batch.
                     match self.encode_type {
-                        EncodeType::TypeDefault => {
-                            data.reserve(result.physical_columns.maximum_encoded_size(
+                        EncodeType::TypeArrow => {
```

Comment:
How about:

```rust
let response_encode_type;
match self.desired_encode_type {
    EncodeType::TypeArrow => {
        response_encode_type = EncodeType::TypeArrow;
        ....
    }
    _ => {
        // For the default or unsupported encode type, use datum format.
        response_encode_type = EncodeType::TypeDefault;
    }
}
...
sel_resp.set_encode_type(response_encode_type);
```

We will only need to change code in one place when a new encoding is added.

Reply:
Ok. I update the code. PTAL again.